app/models/invoice.rb

```diff
@@ -11,6 +11,10 @@ class Invoice < ApplicationRecord
 
   before_save :ensure_organization_sequential_id, if: -> { organization.per_organization? }
   before_save :ensure_number
+  after_create :trigger_invoice_sync, if: :finalized?
+  after_update :trigger_invoice_sync, if: -> { status_updated_to_finalized? }
+  after_create :trigger_sales_order_sync, if: :finalized?
+  after_update :trigger_sales_order_sync, if: -> { status_updated_to_finalized? }
```

Comment:
I think that we can use just `after_save` here and use the same check that we have for setting invoice number (`status_changed_to_finalized?`). Invoice number is also set only when the invoice is finalized

Reply:
The check was not working in after_save:

```
rspec ./spec/models/invoice_spec.rb:327 # Invoice#save with an existing record when status changed to finalized with integration customer when sync invoices is true when sync sales orders is true enqueues invoice create job
rspec ./spec/models/invoice_spec.rb:331 # Invoice#save with an existing record when status changed to finalized with integration customer when sync invoices is true when sync sales orders is true enqueues sales order create job
rspec ./spec/models/invoice_spec.rb:339 # Invoice#save with an existing record when status changed to finalized with integration customer when sync invoices is true when sync sales orders is false enqueues invoice create job
rspec ./spec/models/invoice_spec.rb:359 # Invoice#save with an existing record when status changed to finalized with integration customer when sync invoices is false when sync sales orders is true enqueues sales order create job
```

Reply:
In case of invoice number it is in the before_save calback:

```
before_save :ensure_number
```